mne/realtime/tests/test_mockclient.py

```diff
@@ -29,16 +31,44 @@ def test_mockclient():
     data = epochs.get_data()
 
     rt_client = MockRtClient(raw)
+    # choose "large" value, should always be longer than execution time of
+    # get_data()
+    isi_max = 5
     rt_epochs = RtEpochs(rt_client, event_id, tmin, tmax, picks=picks,
-                         isi_max=0.5)
+                         isi_max=isi_max)
 
     rt_epochs.start()
     rt_client.send_data(rt_epochs, picks, tmin=0, tmax=10, buffer_size=1000)
 
+    # get_data() should return immediately and not wait for the timeout
+    start_time = time.time()
     rt_data = rt_epochs.get_data()
-
+    retrieval_time = time.time() - start_time
+    assert_true(retrieval_time < isi_max)
     assert_true(rt_data.shape == data.shape)
     assert_array_equal(rt_data, data)
+    assert_true(len(rt_epochs) == len(epochs))
+
+    # iteration over epochs should block until timeout
+    rt_iter_data = list()
+    start_time = time.time()
+    for cur_epoch in rt_epochs:
+        rt_iter_data.append(cur_epoch)
+    retrieval_time = time.time() - start_time
+    assert_true(retrieval_time >= isi_max)
```

Comment:
I added the timeout check to make sure behavior is consistent with the previous implementation. I think, waiting for ``0.5s`` would be long enough to make sure the delay is not due to a slow build environment. But I could also remove the timing part if it takes too long/seems too fragile to you